### PR TITLE
[Core] Load plugins safely

### PR DESCRIPTION
**Pull request by vurusovs**
### Details:
 - To prevent any security issues (like injected library loading via environment variables etc.) all plugins should be loaded by absolute file path. Extensions mechanism is left as is in order to provide more flexibility and don't break existed behavior
 - For greater reliability, algorithm of plugins searching was changed:
   - if provided path is absolute, return it for both `register_plugin()` and `plugins.xml` cases
   - if path is relative, it will be treated as relative to working directory for `register_plugin()` and relative to library directory (libopenvino.so place) for `plugins.xml`
   - If only name is provided (`register_plugin()` case), it will be searched in library directory (not as relative)
   
    In all cases algorithm will provide only absolute paths to comply with p.1
- For testing purposes `getPluginPath*()` were moved to separate .hpp and .cpp files

### Tickets:
 - 96201, 97906, 97899

### CODE REVIEW

File: src/inference/include/openvino/runtime/core.hpp [hunk 1]
```diff
@@ -604,9 +604,9 @@ class OPENVINO_RUNTIME_API Core {
      * and prefix to identify library full name.
      * For example, on Linux platform, plugin name specified as `plugin_name` will be wrapped as `libplugin_name.so`.
      * Plugin search algorithm:
-     * - If plugin is located in the same directory as OpenVINO runtime library, it will be used.
-     * - If no, plugin is tried to be loaded from paths pointed by PATH/LD_LIBRARY_PATH/DYLD_LIBRARY_PATH
-     *   environment variables depending on the platform.
+     * - If `plugin_name` points to an exact library path (absolute or relative), and file exists, it will be used.
```
Comment (vurusovs) [inline src/inference/include/openvino/runtime/core.hpp:607]: Update description with correct behavior

File: src/inference/src/ie_core_get_plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
```
Comment (vurusovs) [inline src/inference/src/ie_core_get_plugin.cpp:5]: remove

File: src/common/transformations/tests/common_optimizations/transpose_sinking_general_test.cpp [hunk 3]
```diff
@@ -2,18 +2,14 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include <transformations/common_optimizations/transpose_sinking_general.hpp>
-
-#include <transformations/init_node_info.hpp>
+#include <functional>
```
Comment (ilyachur) [inline src/common/transformations/tests/common_optimizations/transpose_sinking_general_test.cpp:5]: Looks like you change nothing in this file. Could you revert changes?

**Review by ilyachur — COMMENTED** [review]
Can we be sure that this PR won't affect our users?

File: src/common/util/src/file_util.cpp [hunk 4]
```diff
@@ -348,17 +353,45 @@ std::wstring ov::util::string_to_wstring(const std::string& string) {
...
     absolutePath.resize(MAX_ABS_PATH);
-    auto absPath = get_absolute_path(&absolutePath[0], path);
-    if (!absPath) {
-        std::stringstream ss;
-        ss << "Can't get absolute file path for [" << path << "], err = " << strerror(errno);
-        throw std::runtime_error(ss.str());
+    if (strict) {
+        char* absPath;
+        // canonicalize path and check file existence and accessibility
+        absPath = get_absolute_path(&absolutePath[0], path);
```
Comment (vurusovs) [inline src/common/util/src/file_util.cpp:362]: Use `get_absolute_path`, even if `absPath` is NULL (file doesn't exist, on Linux), `absolutePath` will contain concatenated path

**Comment by vurusovs** [comment]
Work will be finished in scope of https://github.com/openvinotoolkit/openvino/pull/15073

**Pull request closed without merging**